grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspace.java

```diff
@@ -98,6 +98,9 @@ public boolean ensureKeyspaceInitialised(String keyspace) {
 
         try (GraknGraph graph = factory.getGraph(SYSTEM_GRAPH_NAME, GraknTxType.WRITE)) {
             ResourceType<String> keyspaceName = graph.getOntologyConcept(KEYSPACE_RESOURCE);
+            if (keyspaceName == null) {
+                throw new RuntimeException("Graph for keyspace [" + keyspace + "] not properly initialized. Missing keyspace name resource");
```

Comment:
This is the race condition I was dealing with at some point. I'll change it to GraknException but it's nicer to have a proper message there rather than a nullpointer exception